test/src/cache/test_utils.dart

```diff
@@ -28,5 +29,34 @@ Future<void> cleanUpIntegrationTests(TestSetup testSetup) async {
 }
 
 String generateStringWithUuid(String prefix) {
-  return "$prefix-${uuidGenerator.v4()}";
+  return "$prefix-${uuidGenerator.v4()}-dart-int-test";
+}
+
+Future<void> verifyListFetch(String cacheName, String listName,
+    List<Value> expected, CacheClient cacheClient) async {
+  final listResp = await cacheClient.listFetch(cacheName, listName);
+  switch (listResp) {
+    case ListFetchHit():
+      expect(
+          listResp.values.toList(), expected.map((e) => e.toUtf8()).toList());
+      break;
+    case ListFetchMiss():
+      fail('Expected Hit but got Miss');
+    case ListFetchError():
+      fail('Expected Hit but got Error');
```

Comment:
`fail` statements in helper functions should also include error code and message